public/network-realms.mjs

```diff
@@ -838,12 +842,8 @@ class VirtualIrc {
   link(realm) {
     const {networkedIrcClient} = realm;
 
-    // note: this is not a good place for this, since it doesn't have to do with players
-    // it's here for convenience
     const onchat = e => {
-      this.parent.dispatchEvent(new MessageEvent('chat', {
-        data: e.data,
-      }));
+      this.parent.receivedChatMessage(e.data);

```

Comment:
This makes the code worse than it was, because now there is a tight coupling with the new method. And this is _still_ not the right place for the trigger, while the comment warning of this fact was removed...

Reply:
Changed reverted (not included) in https://github.com/webaverse/multiplayer-do/pull/6